sdk/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java

```diff
@@ -159,24 +160,31 @@ static RecordEventsReadableSpan startSpan(
   @Override
   public SpanData toSpanData() {
     SpanContext spanContext = getSpanContext();
-    return SpanData.newBuilder()
-        .setName(getName())
-        .setInstrumentationLibraryInfo(instrumentationLibraryInfo)
-        .setTraceId(spanContext.getTraceId())
-        .setSpanId(spanContext.getSpanId())
-        .setTraceFlags(spanContext.getTraceFlags())
-        .setTracestate(spanContext.getTracestate())
-        .setAttributes(getAttributes())
-        .setStartEpochNanos(startEpochNanos)
-        .setEndEpochNanos(getEndEpochNanos())
-        .setKind(kind)
-        .setLinks(getLinks())
-        .setParentSpanId(parentSpanId)
-        .setHasRemoteParent(hasRemoteParent)
-        .setResource(resource)
-        .setStatus(getStatus())
-        .setTimedEvents(adaptTimedEvents())
-        .build();
+    // Set immutable info outside the lock.
+    SpanData.Builder builder =
+        SpanData.newBuilder()
+            .setInstrumentationLibraryInfo(instrumentationLibraryInfo)
+            .setTraceId(spanContext.getTraceId())
+            .setSpanId(spanContext.getSpanId())
+            .setTraceFlags(spanContext.getTraceFlags())
+            .setTracestate(spanContext.getTracestate())
+            .setStartEpochNanos(startEpochNanos)
+            .setKind(kind)
+            .setLinks(getLinks())
+            .setParentSpanId(parentSpanId)
+            .setHasRemoteParent(hasRemoteParent)
+            .setResource(resource);
+    synchronized (lock) {
+      builder
+          .setHasEnded(hasEnded)
+          .setName(getName())
+          .setAttributes(hasEnded ? attributes : new HashMap<>(attributes))
```

Comment:
why do we change this behavior?

Reply:
That was a mistake: https://github.com/open-telemetry/opentelemetry-java/issues/699. I'll revert that part.

Reply:
Can you add a simple test that all components are set for active/ended spans?

Reply:
What do you mean? Like checking that all span data fields are properly set? I could do that but this is unrelated to this line (even though the `new HashMap` is useless, it would not cause any unit tests to fail, the attributes are still properly set for both ended and un-ended Spans).

Reply:
I added assertions for the two new properties of SpanData to the unit tests (425edcc, a030e1b). I still think that latencyNanos should not be part of SpanData, mostly because it makes unit testing with active spans a pain.